sql-cli/sql_cli/dag_generator.py

```diff
@@ -80,17 +80,16 @@ def sorted_sql_files(self) -> list[SqlFile]:
         return list(depth_first_search.dfs_postorder_nodes(graph))
 
 
-def generate_dag(directory: Path, target_directory: Path, dags_directory: Path) -> Path:
+def generate_dag(directory: Path, dags_directory: Path) -> Path:
     """
     Generate a DAG from SQL files.
 
     :params directory: The directory containing the raw sql files.
-    :params target_directory: The directory containing the executable sql files.
     :params dags_directory: The directory containing the generated DAG.
 
     :returns: the path to the DAG file.
     """
-    sql_files = sorted(get_sql_files(directory, target_directory))
+    sql_files = sorted(get_sql_files(directory, target_directory=dags_directory / ".sql"))
```

Comment:
Why do you sort these?

Reply:
Because it is a `set` of sql_files. A SqlFile has a unique path. You cannot have the same file twice. I found it more logical to return a set. But happy to change it to a list. However I am not sure if the `glob` (used underneath) is deterministic (see its [documentation](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob)). In the example they actually also sort the files after.

Reply:
If we use a non-deterministic structure like set for generating the tasks in the DAG, we will have a non-deterministic order of tasks if they have no dependencies for example.